### PR TITLE
[Whisper] Fix decoder ids methods

### PR DESCRIPTION
**Pull request by sanchit-gandhi**
# What does this PR do?

The previous PR https://github.com/huggingface/transformers/pull/20589 incorrectly returned a list of forced decoder ids:
```python
from transformers import WhisperProcessor

processor = WhisperProcessor.from_pretrained("openai/whisper-tiny.en")
print(processor.get_decoder_prompt_ids(task="transcribe"))
```
**Print Output:**
```
[50257, 50358, 50362]
```

The correct format is a nested list of decoder ids, where the first element of each list specifies the position of the forced token and the second the token id:
```python
print(processor.get_decoder_prompt_ids(task="transcribe"))
```
**Print Output:**
```
[(1, 50257), (2, 50358), (3, 50362)]
```
(at position 1 we force token 50257, at 2 we force 50358, at 3 we force 50362)

The PR also implements a test, thus making sure that no such error can be made again 😅

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/whisper/tokenization_whisper.py [hunk 1]
```diff
@@ -583,5 +583,6 @@ def _build_conversation_input_ids(self, conversation) -> List[int]:
         return input_ids
 
     def get_decoder_prompt_ids(self, task=None, language=None, no_timestamps=True):
-        self.set_prefix_tokens(task=task, language=language, predict_timestamps=no_timestamps)
-        return self.prefix_tokens
+        self.set_prefix_tokens(task=task, language=language, predict_timestamps=not no_timestamps)
+        forced_decoder_ids = [(rank + 1, token) for rank, token in enumerate(self.prefix_tokens)]
+        return forced_decoder_ids
```
Comment (ArthurZucker) [inline src/transformers/models/whisper/tokenization_whisper.py:588]: Nice catch! I should have realized when reviewing!

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

File: tests/models/whisper/test_processor_whisper.py [hunk 2]
```diff
@@ -128,3 +133,17 @@ def test_model_input_names(self):
             feature_extractor.model_input_names,
             msg="`processor` and `feature_extractor` model input names do not match",
         )
+
+    def test_get_decoder_prompt_ids(self):
```
Comment (ArthurZucker) [inline tests/models/whisper/test_processor_whisper.py:137]: Nice!

**Review by ArthurZucker — APPROVED** [review]
Thanks for the quick fix

**Pull request merged**